src/mongocxx/test/spec/unified_test_format.cpp

```diff
@@ -113,6 +111,180 @@ bool has_run_on_requirements(const bsoncxx::document::view test) {
     return std::any_of(std::begin(requirements), std::end(requirements), compatible_with_server);
 }
 
+std::string json_kvp_to_uri_kvp(std::string s) {
+    // The transformation is as follows:
+    //     1. "{ "key" : "value" }"     -- initial representation
+    //     2. "key:value"               -- intermediate step (without quotes)
+    //     3. "key=value"               -- final step (without quotes)
+
+    using namespace std;
+    auto should_remove = [&](const char c) {
+        const auto remove = {' ', '"', '{', '}'};
+        return end(remove) != find(begin(remove), end(remove), c);
+    };
+
+    s.erase(remove_if(begin(s), end(s), should_remove), end(s));
+    replace(begin(s), end(s), ':', '=');
+    return s;
+}
+
+std::string json_to_uri_opts(const std::string& input) {
+    // Transforms a non-nested JSON document string (assumed to contain URI keys and values) to a
+    // string of equivalent URI options and values. That is,
+    //      input   := "{ "readConcernLevel" : "local", "w" : 1 }"
+    //      output  := "readConcernLevel=local&w=1"
+    std::vector<std::string> output;
+    const std::regex delim(",");
+    std::transform(std::sregex_token_iterator(std::begin(input), std::end(input), delim, -1),
+                   std::sregex_token_iterator(),
+                   std::back_inserter(output),
+                   json_kvp_to_uri_kvp);
+
+    auto join = [](const std::string& s1, const std::string& s2) { return s1 + "&" + s2; };
+    return std::accumulate(std::begin(output) + 1, std::end(output), output[0], join);
+}
+
+std::string uri_options_to_string(document::view object) {
+    // Spec: Optional object. Additional URI options to apply to the test suite's connection string
+    // that is used to create this client. Any keys in this object MUST override conflicting keys in
+    // the connection string.
+    if (!object["uriOptions"])
+        return {};
+
+    // TODO: Spec: if 'readPreferenceTags' is specified in this object, the key will map to an array
```

Comment:
Nice, it's good that this is called out with an explicit comment. If this does appear in a future test, I think `readPreferenceTags` and `readPreference` could be handled as a special case and removed from input `object` before converting the rest to string with `json_to_uri_opts`.

Reply:
Noted.